### PR TITLE
3515 docs glossary

### PR DESCRIPTION
**Pull request by vinodhini8694**
Added relative links to enable glossary to work with different versions.

### CODE REVIEW

File: docs/glossary.md [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+
+### Main Branch
+Every Git repository has the main branch (unless you take explicit steps to remove it) and it plays a key role in the software development process. In most projects, it represents the source of truth - all the code that works has been tested and is ready to be pushed to production. Similarly, main branch in lakeFS could be used as the single source of truth. For example, the live production data can be on the main branch. 
+
+### Metadata Management
+Where there is data, there is also metadata. lakeFS uses metadata to define schema, data types, data versions, relations to other datasets, etc. This helps to improve discoverability and manageability. lakeFS performs data versioning through metadata operations. 
+
+### Merge
+In lakeFS, once you commit data, you can review it and then merge the committed data into the target branch. A merge generates a commit on the target branch with all your changes. lakeFS guarantees atomic merges that are fast, given they don’t involve copying data.
```
Comment (arielshaqed) [inline docs/glossary.md:70]: We should define "merge" here.

**Review by arielshaqed — APPROVED** [review]
Thanks!

This duplicates parts of the "object model" and "branching model" pages.  I am unsure what goes in which page.  This is important now (to reduce duplicated paragraphs), but especially going forward (when contributors and developers will need to decide where to put documentation).

File: docs/glossary.md (hunk 1 above)
Comment (vinodhini8694) [reply docs/glossary.md:70]: lakeFS merge command, similar to git merge functionality, allows you to merge data branches. Once you commit data, you can review it and then merge the committed data into the target branch. A merge generates a commit on the target branch with all your changes. lakeFS guarantees atomic merges that are fast, given they don’t involve copying data.

File: docs/glossary.md [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+
+### Commit
+A commit is a point-in-time immutable snapshot of a branch. It's a collection of object metadata and data, including paths and the object contents and metadata. Commits have their own commit metadata. Every repository has one initial commit with no parent commits. If a commit has more than one parent, it is a merge commit. lakeFS supports only merge commits with two parents.
+
+### Cross-collection Consistency
+It is unfortunate that the word 'consistency' has multiple meanings, at least four of them according to Martin Kleppmann. Consistency in the context of lakeFS and data versioning is, the guarantee that operations in a transaction are performed accurately, correctly and most important, atomically. 
+
+A repository (and thus a branch) in lakeFS, can span multiple tables or collections. By providing branch, commit, merge and revert operations atomically on a branch, lakeFS achieves consistency guarantees across different logical collections. That is, data versioning is consistent across multiple collections within a repository.
+
+It is sometimes referred as multi-table transactions. That is, lakeFS offers transactional guarantees across multiple tables.
```
Comment (vinodhini8694) [inline docs/glossary.md:37]: @arielshaqed Is this definition of cross-collection consistency clear?

**Pull request merged**